resolver/resolver.cc

```diff
@@ -1377,6 +1377,11 @@ class ResolveSignaturesWalk {
                         return;
                     }
                 }
+                if (ast::isa_tree<ast::UnresolvedConstantLit>(expr) || ast::isa_tree<ast::ConstantLit>(expr)) {
+                    // we don't want to report an error here because constants that are aliases for other constants can
```

Comment:
I didn't understand this special case.

Reply:
I was actually in the process of writing an explanation as you were looking at this: https://github.com/sorbet/sorbet/pull/1572#discussion_r313622027